multiqc/modules/mosdepth/mosdepth.py

```diff
@@ -243,60 +272,60 @@ def parse_cov_dist(self):
                         if len(include_contigs) > 0 and not any(
                             fnmatch.fnmatch(contig, pattern) for pattern in include_contigs
                         ):
-                            # Commented out since this could be many thousands of contigs fo reach!
-                            # log.debug("Skipping not included contig '{}'".format(contig))
+                            # Commented out since this could be many thousands of contigs!
+                            # log.debug(f"Skipping not included contig '{contig}'")
                             continue
 
                         avg = perchrom_avg_data[s_name].get(contig, 0) + float(bases_fraction)
                         perchrom_avg_data[s_name][contig] = avg
 
-                if s_name in dist_data:
+                if s_name in cumcov_dist_data:
                     self.add_data_source(f, s_name=s_name, section="genome_results")
 
-        return dist_data, cov_data, xmax, perchrom_avg_data
+            # Correct per-contig average, since mosdepth reports cumulative coverage for at least
+            # a certain value (see https://github.com/brentp/mosdepth#distribution-output).
+            # For that reason, the 0 category (which is always 1) should not be included.
+            for i in perchrom_avg_data:
+                for j in perchrom_avg_data[i]:
+                    perchrom_avg_data[i][j] -= 1
```

Comment:
Hi @ewels, thanks for merging!

However, I am a bit puzzled why it comes out as a negative value. On that example you sent, `multiqc` should sum, for each chromosome, across all positions:



20 * 0.01

Reply:
I've got it.. This code loop runs twice (`for scope in ("region", "global"):`) for each file, but `perchrom_avg_data` is defined and added to outside of the loop. So it subtracts `1` correctly for `region` and then subtracts a second time for `global`.

If we just drop the indentation by one for this code block so that it's outside of that loop, then it should be fine.

Reply:
Fixed in 471bd379d57e99625a83fe1618753274a812877a

Reply:
ahh, makes sense! thanks for figuring it out! 👍